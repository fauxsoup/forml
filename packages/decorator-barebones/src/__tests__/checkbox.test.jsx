import Checkbox from "../checkbox";
import Context from "@forml/context";
import React from "react";
import { render } from "@testing-library/react";
import * as decorator from "../";

describe("renders", function() {
    let form;
    let title = "title";
    let description = "description";
    let styles;
    let root;
    let header;
    let icon;

    beforeEach(function() {
        form = { type: "checkbox", key: [] };
    });

    test("with no title or description", function() {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <Checkbox form={form} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test("with title but no description", function() {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <Checkbox form={form} title={title} />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test("when checked", function() {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <Checkbox
                    form={form}
                    title={title}
                    description={description}
                    checked={true}
                />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });
});
