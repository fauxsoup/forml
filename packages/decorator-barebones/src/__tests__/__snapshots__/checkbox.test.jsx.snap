// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when checked 1`] = `
<div>
  <div>
    <label>
      <input
        checked=""
        type="checkbox"
      />
       
      title
    </label>
    <p>
      description
    </p>
  </div>
</div>
`;

exports[`renders with no title or description 1`] = `
<div>
  <div>
    <label>
      <input
        type="checkbox"
      />
       
    </label>
  </div>
</div>
`;

exports[`renders with title but no description 1`] = `
<div>
  <div>
    <label>
      <input
        type="checkbox"
      />
       
      title
    </label>
  </div>
</div>
`;
